File: mycollab-dao/src/main/java/com/esofthead/mycollab/security/BooleanPermissionFlag.java
/**
 * This file is part of mycollab-dao.
 *
 * mycollab-dao is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-dao is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-dao.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.esofthead.mycollab.security;

import com.esofthead.mycollab.common.i18n.SecurityI18nEnum;

/**
 * Boolean permission flag
 *
 * @author MyCollab Ltd
 * @version 1.0
 */
public class BooleanPermissionFlag extends PermissionFlag {
    public static final int TRUE = 128;
    public static final int FALSE = 129;

    /**
     * Check whether <code>flag</code> is true permission
     *
     * @param flag
     * @return
     */
    public static boolean beTrue(Integer flag) {
        return (flag == TRUE);
    }

    /**
     * Check whether <code>flag</code> is false permission
     *
     * @param flag
     * @return
     */
    public static boolean beFalse(Integer flag) {
        return (flag == FALSE);
    }

    public static SecurityI18nEnum toKey(Integer flag) {
        return (flag == TRUE) ? SecurityI18nEnum.YES : SecurityI18nEnum.NO;
    }
}

File: mycollab-dao/src/main/java/com/mycollab/security/BooleanPermissionFlag.kt
/**
 * Copyright © MyCollab
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.security

import com.mycollab.common.i18n.SecurityI18nEnum

/**
 * Boolean permission flag
 *
 * @author MyCollab Ltd
 * @version 1.0
 */
class BooleanPermissionFlag : PermissionFlag() {
    companion object {
        @JvmField val TRUE = 128

        @JvmField val FALSE = 129

        /**
         * Check whether `flag` is true permission
         *
         * @param flag
         * @return
         */
        fun beTrue(flag: Int?): Boolean = flag == TRUE

        /**
         * Check whether `flag` is false permission
         *
         * @param flag
         * @return
         */
        fun beFalse(flag: Int?): Boolean = flag == FALSE

        fun toKey(flag: Int?): SecurityI18nEnum = if (flag == TRUE) SecurityI18nEnum.YES else SecurityI18nEnum.NO
    }
}
